### DISCUSSION TITLE
kms_grant_v1: "key_id = var.kms_id"

### DISCUSSION THREAD

**Discussion by rramge** [opentelekomcloud/terraform-provider-opentelekomcloud#1803] [Q&A]
Hi,

I don't understand the exact use case of this resource, based on the documentation. And there is also no example for it in the provider's example directory. The given example in the documentation is:

```
resource "opentelekomcloud_kms_grant_v1" "grant_1" {
  key_id            = var.kms_id
  name              = "my_grant"
  grantee_principal = var.user_id
  operations        = ["describe-key", "create-datakey", "encrypt-datakey"]
}
```
with documentation for key_id:

> [key_id](https://registry.terraform.io/providers/opentelekomcloud/opentelekomcloud/latest/docs/resources/kms_grant_v1#key_id) - (Required) Indicates the ID of the KMS. Changing this creates new grant.

Does it now required a key ID (name of argument) or the ID of the entire KMS?
IF its the latter, how can the KMS ID be retrieved? I didn't find a datasource for it.

**Long story short: Can you please give me a working example for using this resource?**

Thanks,

Ralf

**Comment by rramge** [comment]
Hello,

can anyone please help with this? Or shall I open an issue?


KR,

Ralf
